zulip/zulip
# Add filter to direct messages list.

Fixes https://github.com/zulip/zulip/issues/22113.

The search will only be visible when in the `more conversations`
view. Once we click `back to channels` and close the
`more conversations` view, the search will clear and the search
box will disappear.

We've chosen `pm_list_data.get_conversations` as the function
to which we will pass our search term. We could have technically
found the value of the filter element via JQuery in pm_list_data,
but pm_list_data does not handle any JQuery and we should keep
it that way.

`pm_list_data.get_list_info` also accepts the search_term so that
the info it returns in expanded view is accurate.



**Screenshots and screen captures:**

<details>
<summary>Filter is hidden when more conversations is not selected </summary>
<img width="410" alt="Screenshot 2024-06-06 at 3 19 19 PM" src="https://github.com/zulip/zulip/assets/13910561/5016376e-fe16-4c71-876a-75257911ce99">

</details>


<details>
<summary>Filter with the query `des` </summary>
<img width="343" alt="Screenshot 2024-06-06 at 3 19 51 PM" src="https://github.com/zulip/zulip/assets/13910561/e2df80aa-89e9-49a7-8ba0-913345d6dbef">

</details>

<details>
<summary>Blue background when the DM section is selected</summary>
<img width="355" alt="Screenshot 2024-06-06 at 3 20 06 PM" src="https://github.com/zulip/zulip/assets/13910561/efeb91bf-38d1-4a33-b267-d42bce90b6fb">

</details>

Manual testing checklist:
- Clicking `back to channels` should clear the search and update the message accordingly.
- Clicking on the `x` icon on the filter input box should clear the search and update the list.

Regression testing:
- Test channel list filter and topic list filter for regressions. 

Self review checklist: 
<details>
<summary>Self-review checklist</summary>





      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.


Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).


Completed manual review and testing of the following:

</details>
